18F/identity-idp
# LG-8771: Implement reCAPTCHA on phone setup screens

## 🎫 Ticket

[LG-8771](https://cm-jira.usa.gov/browse/LG-8771)

## 🛠 Summary of changes

Adds initial verification implementation for reCAPTCHA to phone setup screens.

## 📜 Testing Plan

Requires some setup:

1. Create a new reCAPTCHA site for use in your local environment: https://www.google.com/recaptcha/admin/create
2. Assign credentials as `recaptcha_site_key` and `recaptcha_secret_key` in `config/application.yml`
3. Set a score threshold to a float between `0.0` and `1.0` for `phone_setup_recaptcha_score_threshold`

Testing procedure:

1. Go to http://localhost:3000
5. Sign up for new account
6. Choose phone as MFA
7. Enter a phone number (international or U.S.)
8. Submit
9. Confirm that associated reCAPTCHA event is logged if having entered a non-US number†
10. Proceed to next step only if you're successful in reCAPTCHA validation

Log verification: `tail -n100 log/events.log | jq 'select(.name | test("reCAPTCHA.*")) | .'`

Example:

```json
{
  "name": "reCAPTCHA verify result received",
  "properties": {
    "event_properties": {
      "recaptcha_result": {
        "success": true,
        "challenge_ts": "2023-02-07T21:34:03Z",
        "hostname": "localhost",
        "score": 0.9,
        "action": "phone_setup"
      },
      "score_threshold": 0.6,
      "evaluated_as_valid": true
    }
  }
}
```
